modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/TranscriptsManagerImpl.kt

```diff
@@ -33,10 +48,50 @@ class TranscriptsManagerImpl @Inject constructor(
         }
         return availableTranscripts.firstOrNull()
     }
+
+    override suspend fun loadTranscript(
+        url: String,
+        source: LoadTranscriptSource,
+        forceRefresh: Boolean,
+    ) = withContext(Dispatchers.IO) {
+        try {
+            var response = if (forceRefresh && networkWrapper.isConnected()) {
+                service.getTranscript(url, CacheControl.FORCE_NETWORK)
+            } else {
+                service.getTranscript(url, CacheControl.parse(Headers.headersOf("Cache-Control", "only-if-cached, max-stale=7776000")))
+            }
+            if (response.isSuccessful) {
+                response.body()
+            } else {
+                if (!networkWrapper.isConnected()) {
+                    throw NoNetworkException()
+                } else {
+                    response = service.getTranscript(url, CacheControl.FORCE_NETWORK)
+                    if (response.isSuccessful) {
+                        response.body()
+                    } else {
+                        LogBuffer.e(LogBuffer.TAG_INVALID_STATE, "Failed to load transcript from $url: ${response.errorBody()}")
+                        response.errorBody()
+                    }
+                }
+            }
+        } catch (e: UnknownHostException) {
+            LogBuffer.e(LogBuffer.TAG_INVALID_STATE, "Failed to load transcript from $url", e)
+            if (source == LoadTranscriptSource.DOWNLOAD_EPISODE) null else throw NoNetworkException() // fail silently if loaded as part of episode download
+        } catch (e: Exception) {
+            LogBuffer.e(LogBuffer.TAG_INVALID_STATE, "Failed to load transcript from $url", e)
+            if (source == LoadTranscriptSource.DOWNLOAD_EPISODE) null else throw e // fail silently if loaded as part of episode download
+        }
+    }
 }
 
 enum class TranscriptFormat(val mimeType: String) {
     SRT("application/srt"),
     VTT("text/vtt"),
     HTML("text/html"),
 }
+
+enum class LoadTranscriptSource {
+    DOWNLOAD_EPISODE, // When transcript is downloaded as part of episode download
+    DEFAULT,
+}
```

Comment:
I think this should be rather defined next to the interface and not next to the implementation.

Reply:
Moved to interface in e13252bd493f70b6dfb038df9cbf0d35b922d1aa